### DISCUSSION TITLE
I am confused!!

### DISCUSSION THREAD

**Discussion by kartik-giri** [Cyfrin/foundry-full-course-cu#1327] [Q&A]
hey guys is this possible to call smart contract function through post Api and make blockchain state change using post api. please answer!

**Comment by ghost** [comment]
Yes you can. Every Ethereum client implements a series of JSON RPC API with different languages. JSON RPC API define structures to interact with Etheruem (not just smart contract).

We can fetch a block using `eth_getBlockByNumber` method with JavaScript using the `fetch` API.

```javascript
const MAINNET_RPC_URL = "your_rpc_url"
const BLOCK_NUMBER = "block_number"

export const getBlock = async () => {
    const headers = new Headers()
    headers.append("Content-Type", "application/json")

    const body = JSON.stringify({
        jsonrpc: "2.0",
        id: 0,
        method: "eth_getBlockByNumber",
        params: [BLOCK_NUMBER, false],
    })

    const requestOptions = {
        method: "POST",
        headers,
        body,
    }

    const res = await fetch(MAINNET_RPC_URL, requestOptions)

    if (!res.ok) {
        throw new Error("Failed to fetch block")
    }

    const block = await res.json()
    return block.result
}
```

Usually we don't want to use JSON-RPC directly because there are libraries or wrappers that simplify everything. Check ethers.js or web3.py.
